File: app/src/main/java/com/xomena/cmpfutboltfe/MainPagerAdapter.java
package com.xomena.cmpfutboltfe;
import com.xomena.cmpfutboltfe.model.*;
import com.xomena.cmpfutboltfe.util.*;

import android.content.Context;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class MainPagerAdapter extends FragmentPagerAdapter {
    final int PAGE_COUNT = 2;
    private int tabTitles[] = new int[] { R.string.select_county, R.string.route_map };
    private Context context;

    public MainPagerAdapter(FragmentManager fragmentManager, Context context) {
        super(fragmentManager);
        this.context = context;
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return CountiesFragment.newInstance();
            case 1:
                return MainMapFragment.newInstance();
            default:
                return null;
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        // Generate title based on item position
        return context.getString(tabTitles[position]);
    }
}

File: app/src/main/java/com/xomena/cmpfutboltfe/ui/adapter/MainPagerAdapter.kt
package com.xomena.cmpfutboltfe.ui.adapter

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.xomena.cmpfutboltfe.MainMapFragment
import com.xomena.cmpfutboltfe.R
import com.xomena.cmpfutboltfe.ui.fragment.CountiesFragment

@Suppress("DEPRECATION")
class MainPagerAdapter(
    fragmentManager: FragmentManager,
    private val context: Context
) : FragmentPagerAdapter(fragmentManager) {

    private val tabTitles = intArrayOf(R.string.select_county, R.string.route_map)

    override fun getCount(): Int = PAGE_COUNT

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> CountiesFragment.newInstance()
            1 -> MainMapFragment.newInstance()
            else -> CountiesFragment.newInstance() // Default fallback
        }
    }

    override fun getPageTitle(position: Int): CharSequence {
        return context.getString(tabTitles[position])
    }

    companion object {
        private const val PAGE_COUNT = 2
    }
}
